### PR TITLE
Share a single MongoClient instance

### LINKED ISSUE #193: PanBase creates a PanMongo instance for each created sub-class
This seems likely to be a bad idea.

### PR DESCRIPTION
**Pull request by jamessynge**
Caches clients using a weakref so that the client is discarded if there are no referrers.

Fixes #193.

### CODE REVIEW

File: pocs/utils/database.py [hunk 1]
```diff
@@ -41,6 +57,7 @@ def __init__(self, db='panoptes', host='localhost', port=27017, connect=False, *
             'weather',
         ]
 
+        # Why not: db_handle = self._client.db
```
Comment (wtgee) [inline pocs/utils/database.py]: Good question. Since no comment could probably be changed.

File: pocs/utils/database.py [hunk 2]
```diff
@@ -1,34 +1,50 @@
-import os
-import pymongo
-
+from datetime import date
```
Comment (wtgee) [inline pocs/utils/database.py]: Curious why these were moved to the top and `os` and `pymongo` moved down.

Comment (jamessynge) [reply pocs/utils/database.py]: Putting standard python libraries at the top, then third party, then our own. A style I'm used to... from you know where.
 Probably should discuss whether we should do that globally before making such changes.

Comment (wtgee) [reply pocs/utils/database.py]: I was mostly confused by moving the `os` below the `datetime`. Makes sense for `pymongo`. Not a huge deal.

**Pull request merged**